### PR TITLE
Scion optimizer 

### PR DESCRIPTION
**Pull request by mkhona-nvidia**
Added Scion.

The main change is the parametrization to allow Franke-Wolfe. We now *DO NOT* use weight decay and instead use step size (i.e. the learning rate) and spectral radius, with the unit_rms choice for width scaling

### CODE REVIEW

File: emerging_optimizers/orthogonalized_optimizers/scion.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                logging.error("Triton 3.4.0 or higher is required for use_syrk to be True.")
+                use_syrk = False
+            elif sm_version not in ((8, 0), (9, 0), (10, 0), (10, 3)):
+                logging.error(
+                    f"Correctness of Triton kernel on SM {sm_version} cannot be guaranteed. Setting use_syrk to False."
+                )
+                use_syrk = False
+
+        # Add checks for weight decay arguments to enable Franke-Wolfe step.
+        if weight_decay != 1:
```
Comment (skyw) [inline emerging_optimizers/orthogonalized_optimizers/scion.py]: They should be removed in argument, keep the interface simpler.

File: emerging_optimizers/orthogonalized_optimizers/scion.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+        logging.warning("Scion does not use weight decay. Setting use_independent_wd to False to allow Franke-Wolfe.")
+        use_independent_wd = False
+
+        def scaled_orthogonalize_fn(grad: torch.Tensor) -> torch.Tensor:
+            logging.debug(
+                f"Orthogonalizing grad with {num_ns_steps} steps, {coefficient_type} coefficient, spectral_radius={spectral_radius}"
+            )
+            orth_grad = newton_schulz(grad, steps=num_ns_steps, coefficient_type=coefficient_type, use_syrk=use_syrk)
+            width_factor = (grad.size(-2) / grad.size(-1)) ** 0.5
```
Comment (skyw) [inline emerging_optimizers/orthogonalized_optimizers/scion.py]: Also isn't this spectral scaling? If yes, can just use "get_muon_scale_factor"

File: emerging_optimizers/orthogonalized_optimizers/scion.py (hunk 1 above)
Comment (mkhona-nvidia) [reply emerging_optimizers/orthogonalized_optimizers/scion.py]: done, removed from args and added defaults in `init`

File: emerging_optimizers/orthogonalized_optimizers/scion.py (hunk 2 above)
Comment (mkhona-nvidia) [reply emerging_optimizers/orthogonalized_optimizers/scion.py]: We don't want to specify options here, Scion mathematically needs this scaling only and nothing else

Comment (skyw) [reply emerging_optimizers/orthogonalized_optimizers/scion.py]: you don't need any option, just call that function with fixed argument

Comment (mkhona-nvidia) [reply emerging_optimizers/orthogonalized_optimizers/scion.py]: done

File: emerging_optimizers/orthogonalized_optimizers/scion.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+          [`arXiv:2502.07529 <https://arxiv.org/abs/2502.07529>`_]
+
+    Warning:
+        - This optimizer requires that all parameters passed in are 2D.
+        - It should not be used for the embedding layer, the final fully connected layer, or any 1-D
+          parameters; those should all be optimized by the appropriate LMO for that layer. For example,
+          for 1d params, it is scaled by the `ell_inf` radius.
+
+    Args:
+        {_args_doc}
```
Comment (skyw) [inline emerging_optimizers/orthogonalized_optimizers/scion.py]: This includes argument Scion class doesn't have. Doc should match the class.

File: emerging_optimizers/orthogonalized_optimizers/scion.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                logging.error("Triton 3.4.0 or higher is required for use_syrk to be True.")
+                use_syrk = False
+            elif sm_version not in ((8, 0), (9, 0), (10, 0), (10, 3)):
+                logging.error(
+                    f"Correctness of Triton kernel on SM {sm_version} cannot be guaranteed. Setting use_syrk to False."
+                )
+                use_syrk = False
+
+        # Add checks for weight decay arguments to enable Franke-Wolfe step.
+        logging.warning("Scion does not use weight decay. Setting weight_decay to 1.")
```
Comment (skyw) [inline emerging_optimizers/orthogonalized_optimizers/scion.py]: Change it to info level. Users choose to use this class should know the implication, so it is not a warning to them but infor.

File: emerging_optimizers/orthogonalized_optimizers/scion.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        fp32_matmul_prec: str = "medium",
+        coefficient_type: str = "quintic",
+        num_ns_steps: int = 5,
+        spectral_radius: float = 1.0,
+        use_syrk: bool = False,
+    ) -> None:
+        if num_ns_steps < 1:
+            raise ValueError(f"num_ns_steps must be at least 1, got {num_ns_steps}")
+
+        if use_syrk:
```
Comment (skyw) [inline emerging_optimizers/orthogonalized_optimizers/scion.py]: Recommend to leave out syrk option for this demonstration class for now.

**Review by skyw — CHANGES_REQUESTED** [review]
Some minor changes needed. otherwise LGTM

**Review by skyw — APPROVED** [review]
